x-pack/plugin/mapper-version/src/main/java/org/elasticsearch/xpack/versionfield/VersionStringDocValuesField.java

```diff
@@ -143,8 +144,19 @@ public Version next() {
                 if (hasNext() == false) {
                     throw new NoSuchElementException();
                 }
-                return new Version(getInternal(index++));
+
+                return getVersionFromInternal(index++);
             }
         };
     }
+
+    private Version getVersionFromInternal(int index) {
```

Comment:
Is this needed if we create the Version from a BytesRef directly?

Reply:
Yes, it makes sense, we can do the decoding in the ctor directly. We won't need getVersionFromInternal.
Fixing.

Reply:
I started to do this change and I realised one thing that is worth pointing out: Version(String) constructor is public API (see last Painless test in [20_scripts.yml](https://github.com/elastic/elasticsearch/pull/85990/files#diff-dcc6c3b83691ea8470049bca3dcad374992a030cf25c65bd9e6be3ff0f67c5a6R82)) and it has to encode the string to get the BytesRef. So if we decide to move Version to Server module, we will have to move VersionEncoder as well.